### PR TITLE
set minimum autoray version

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
We need `autoray.ndim`, and it was only introduced in 0.6.1

**Description of the Change:**
- Pin autoray's lower bound to 0.6.1
- update `transmon_interaction` to check if `qubit_freq` is callable before calling `ndim` on it (calling ndim on a function fails with autoray 0.6.1, just returns 0 with 0.6.3, but we shouldn't do it at all because it doesn't quite make sense)

**Benefits:**
This will ensure autoray always works with PennyLane with a standard `pip install pennylane`

**Possible Drawbacks:**
N/A. People have to make their peace with old autoray no longer being supported?

### CODE REVIEW

File: requirements-ci.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ toml
 appdirs
 semantic_version
 dask[delayed]
-autoray
+autoray==0.6.1
```
Comment (timmysilv) [inline requirements-ci.txt:12]:
```suggestion
autoray
```

**Review by lillian542 — APPROVED** [review]
Looks good to me!

File: pennylane/pulse/transmon.py [hunk 2]
```diff
@@ -158,9 +159,9 @@ def transmon_interaction(
         anharmonicity = [0.0] * n_wires
 
     # TODO: make coefficients callable / trainable. Currently not supported
```
Comment (albi3ro) [inline pennylane/pulse/transmon.py:161]: Is this TODO completed?

**Review by albi3ro — APPROVED** [review]

File: pennylane/pulse/transmon.py (hunk 2 above)
Comment (timmysilv) [reply pennylane/pulse/transmon.py:161]: no clue, felt unrelated enough

**Pull request merged**